src/foam/nanos/notification/sms/TwilioSMSService.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.notification.sms',
+  name: 'TwilioSMSService',
+
+  implements: [
+    'foam.nanos.notification.sms.SMSService'
+  ],
+
+  documentation: 'Sends SMS message through Twilio',
+
+  javaImports: [
+    'com.twilio.Twilio',
+    'com.twilio.rest.api.v2010.account.Message',
+    'com.twilio.type.PhoneNumber',
+    'foam.dao.DAO',
+    'foam.nanos.app.TwilioConfig',
+    'foam.nanos.notification.sms.SMSStatus',
+    'foam.util.SafetyUtil'
+  ],
+
+  methods: [
+    {
+      name: 'sendSms',
+      javaCode: `
+        DAO smsMessageDAO = (DAO) x.get("localSmsMessageDAO");
+        TwilioConfig twilioConfig = (TwilioConfig) x.get("twilioConfig");
+        Twilio.init(twilioConfig.getAccountSid(), twilioConfig.getAuthToken());
+
+        String phoneNumber = "";
+        SMSStatus status;
+
+        // check if phone number exists
+        if ( ! SafetyUtil.isEmpty(smsMessage.getPhoneNumber()) ) {
+          phoneNumber = smsMessage.getPhoneNumber();
+        } else {
+          status = SMSStatus.UNSENT;
+          status.setErrorMessage("Phone number not found, cannot send SMS.");
+          smsMessage.setStatus(status);
+          return (SMSMessage) smsMessageDAO.put(smsMessage);
```

Comment:
if we are strictly modelling against the email system, then you shouldn't be putting the smsMessage to the DAO here, but instead you should be letting the cron job or script or calling code put the returned smsMessage to the DAO .. keep this simple and just let it do the sending and set the status and allow the calling code to do the coordination.